src/tests/operators.rs

```diff
@@ -0,0 +1,255 @@
+    assert_eq!(eval_ok("1 >= 1.0"), Value::Bool(true));
+}
+
+#[test]
+fn eval_string_comparison() {
+    assert_eq!(eval_ok("\"abc\" == \"abc\""), Value::Bool(true));
+    assert_eq!(eval_ok("\"abc\" == \"def\""), Value::Bool(false));
+    assert_eq!(eval_ok("\"abc\" != \"def\""), Value::Bool(true));
+    assert_eq!(eval_ok("\"abc\" != \"abc\""), Value::Bool(false));
+    assert_eq!(eval_ok("\"abc\" < \"def\""), Value::Bool(true));
+    assert_eq!(eval_ok("\"abc\" < \"abc\""), Value::Bool(false));
+    assert_eq!(eval_ok("\"abc\" > \"def\""), Value::Bool(false));
+    assert_eq!(eval_ok("\"abc\" > \"abc\""), Value::Bool(false));
+    assert_eq!(eval_ok("\"abc\" <= \"def\""), Value::Bool(true));
+    assert_eq!(eval_ok("\"abc\" <= \"abc\""), Value::Bool(true));
+    assert_eq!(eval_ok("\"abc\" >= \"def\""), Value::Bool(false));
+    assert_eq!(eval_ok("\"abc\" >= \"abc\""), Value::Bool(true));
+}
+
+#[test]
+fn eval_path_string_concatenation() {
+    let curr_dir = std::env::current_dir().unwrap();
+
+    assert_eq!(
+        eval_ok("./hello + \"world\""),
+        Value::Path(format!("{}/helloworld", curr_dir.display()))
+    );
+    assert_eq!(
+        eval_ok("\"hello\" + ./world"),
+        Value::Str(format!("hello{}/world", curr_dir.display()))
+    );
+}
+
+#[test]
+fn eval_path_concat() {
+    assert_eq!(eval_ok(r#"/. + "a""#), Value::Path("/a".to_owned()));
+    assert_eq!(eval_ok(r#"/. + "./a/../b""#), Value::Path("/b".to_owned()));
+}
+
+#[test]
+fn eval_order_of_operations() {
+    // Addition and subtraction have the same precedence, so they should be evaluated from left to right
+    assert_eq!(eval_ok("1 + 2 - 3"), Value::Int(0));
+    assert_eq!(eval_ok("1 - 2 + 3"), Value::Int(2));
+
+    // Multiplication and division have higher precedence than addition and subtraction
+    assert_eq!(eval_ok("1 + 2 * 3"), Value::Int(7));
+    assert_eq!(eval_ok("1 * 2 + 3"), Value::Int(5));
+
+    // Parentheses can be used to change the order of operations
+    assert_eq!(eval_ok("(1 + 2) * 3"), Value::Int(9));
+    assert_eq!(eval_ok("1 * (2 + 3)"), Value::Int(5));
+}
+
+#[test]
+fn eval_string_operator_errors() {
+    assert_eq!(
+        eval_err("1 + \"hello\""),
+        NixErrorKind::TypeMismatch {
+            expected: vec![],
+            got: NixTypeKind::Int
+        }
+    );
+    assert_eq!(
+        eval_err("\"hello\" - \"world\""),
+        NixErrorKind::TypeMismatch {
+            expected: vec![],
+            got: NixTypeKind::String
+        }
+    );
+    assert_eq!(
+        eval_err("\"hello\" * \"world\""),
+        NixErrorKind::TypeMismatch {
+            expected: vec![],
+            got: NixTypeKind::String
+        }
+    );
+    assert_eq!(
+        eval_err("\"hello\" / \"world\""),
+        NixErrorKind::TypeMismatch {
+            expected: vec![],
+            got: NixTypeKind::String
+        }
+    );
+
+    // FIXME: It says string instead of int, because nixjs flips the operator. Should we address?
```

Comment:
It's only a difference in the error message, right? I'd say let's not address until it makes a real-world difference.